src/logic/otp2/controller.ts

```diff
@@ -422,46 +382,74 @@ export async function searchTransit(
             ? metadata.prevDateTime
             : metadata.nextDateTime
 
-        const nextSearchParams: Otp2SearchParams = {
-            ...params,
+        const nextSearchParams = {
+            ...searchParams,
             searchWindow: metadata.searchWindowUsed,
             searchDate: parseISO(dateTime),
         }
-        return searchTransit(nextSearchParams, extraHeaders, queries)
-    }
 
-    if (!tripPatterns.length && !metadata) {
+        return searchTransitUntilMaxRetries(
+            initialSearchDate,
+            nextSearchParams,
+            extraHeaders,
+            queries,
+        )
+    } else {
         const nextMidnight = arriveBy
-            ? endOfDay(subDays(searchDate, 1))
-            : startOfDay(addDays(searchDate, 1))
+            ? endOfDay(subDays(searchDate || new Date(), 1))
```

Comment:
Hm, er `searchDate` potensielt `undefined` her? Det skal vel kanskje ikkje gå an.

Reply:
It's just a consequence of the GetTripPatterns type. I'll update the type as it will never be undefined for us.